src/prog_models/prognostics_model.py

```diff
@@ -458,6 +503,37 @@ def output(self, x) -> dict:
         """
         return {}
 
+    def __output(self, x) -> dict:
+        """
+        Calculate next state, forward one timestep, and add nose
+
+        Parameters
+        ----------
+        x : dict
+            state, with keys defined by model.states \n
+            e.g., x = {'abc': 332.1, 'def': 221.003} given states = ['abc', 'def']
+        
+        Returns
+        -------
+        z : dict
+            Outputs, with keys defined by model.outputs. \n
+            e.g., z = {'t':12.4, 'v':3.3} given outputs = ['t', 'v']
+
+        Example
+        -------
+        | m = PrognosticsModel() # Replace with specific model being simulated
+        | u = {'u1': 3.2}
+        | z = {'z1': 2.2}
+        | x = m.initialize(u, z) # Initialize first state
+        | z = m.__output(3.0, x) # Returns {'o1': 1.2} with noise added
+        """
+
+        # Calculate next state, forward one timestep
+        self.output(x)
+
+        # Add measurement noise
+        return self.apply_measurement_noise(self.output(x))
```

Comment:
Rather than calling self.output(x) twice, try saving it as a variable (z) on the first call